OTP-UI Icons

binh-dam-ibigroup opened the pull request:
This PR removes icons from lib/components/icons and uses those from the OTP-UI icons package instead. The PR also removes unused narrative profile components.

EDITS: Please see the Remarks section below for edits and testing notes.

### Remarks

1. [Outdated as of 9219719bac7f92f1d7ead440e50f9e218947b3d5] Full support of custom icons in OTP-ui is pending opentripplanner/otp-ui#119, however the refactor is mostly compatible with the currently released OTP-ui library.
2. [Outdated as of 9219719bac7f92f1d7ead440e50f9e218947b3d5] In  https://github.com/opentripplanner/otp-react-redux/blob/02fef2765dd684acadfbdcf0d1676853a0cdd4db/example.js#L30, there is a `MyModeIcon` variable passed to various `ModeIcon` props, however `ModeIcon` props will take effect pending a change in `SettingsSelectorPanel` in opentripplanner/otp-ui#119.
3. [Outdated as of 6c10ea651c951d0497289ac8111d7adf867dedc1] As a result, the default icon for "Take Transit" in OTP-RR is changed, and there are subtle visual changes to the itinerary narratives as shown in the screenshots below.
(Note that the bus and walk icon sizes are pending OTP release per https://github.com/opentripplanner/otp-ui/pull/105 to address [this comment](https://github.com/opentripplanner/otp-react-redux/pull/142#pullrequestreview-383985093)) Beware, visual changes taking place! (see screenshots below)
 
![image](https://user-images.githubusercontent.com/56846598/78378792-7ae06e80-759f-11ea-81c9-92d9ee974aa7.png)

![image](https://user-images.githubusercontent.com/56846598/78378867-9186c580-759f-11ea-8ee0-5052d3a6479a.png)

screenshot if using LineItinerary
![image](https://user-images.githubusercontent.com/56846598/78599139-97d6a500-781e-11ea-81ff-3b6119ad6958.png)

4. As of 6c10ea651c951d0497289ac8111d7adf867dedc1, visual styles for trip form elements (not the map or nav bar) match newplanner.trimet.org, and the old styling is abandoned.

evansiroky commented on lib/components/map/default-map.js:
```diff
@@ -13,6 +13,7 @@ import {
   setMapPopupLocation,
   setMapPopupLocationAndGeocode
 } from '../../actions/map'
+
```
No space needed here.

evansiroky commented on lib/components/narrative/default/itinerary-summary.js:
```diff
@@ -1,7 +1,9 @@
 import PropTypes from 'prop-types'
 
-import ModeIcon from '../../icons/mode-icon'
+import TriMetLegIcon from '@opentripplanner/icons/lib/trimet-leg-icon'
```
Should be at the very least grouped with the 3rd party imports and ideally sorted before React and PropTypes. Also PropTypes should be before React.

evansiroky commented:
Just noticed that this PR depends on #147. I'm going to unassign myself until that one gets merged into dev.

binh-dam-ibigroup commented:
There are some components I deleted an need to bring back... I will reassign you when done.

evansiroky commented:
Unassigning myself again until #147 is merged.

landonreed commented on lib/components/form/default-search-form.js:
```diff
@@ -1,10 +1,16 @@
-import React, { Component } from 'react'
+import TriMetModeIcon from '@opentripplanner/icons/lib/trimet-mode-icon'
```
I think we should add a new generic `ModeIcon` to otp-ui that doesn't use the TriMet-specific icons, but instead uses the ones that were originally used in this repo. One reason: we don't want all the rail icons to appear as the Max icon. Another: it is likely confusing to sprinkle in mentions of `TriMet`-branded things in this library, especially with icons which can be closely tied to agency-specific branding.

binh-dam-ibigroup replied:
I agree. We started off with a TriMet-focused approach... How about adding the generic icons and a generic mode-icon/leg-icon class to OTP-ui and isolate the TriMet-specific icons inside OTP-ui?

evansiroky replied:
Seems like a good idea to have a generic library specifically for otp-react-redux since we use it in more regions than just Portland, OR.

binh-dam-ibigroup replied:
The above topic is handled with opentripplanner/otp-ui#119 and the latest commits to this PR.

landonreed commented on lib/components/form/default-search-form.js:
```diff
@@ -1,10 +1,16 @@
+// If no icon is provided for a specific mode, the default OTP-ui icon will be used.
+const customIcons = {
+  TRANSIT: <TriMetModeIcon label='rail' />
+}
```
Default used to be `bus`, right? Why the change?

landonreed commented on lib/components/form/default-search-form.js:
```diff
@@ -13,7 +19,7 @@ export default class DefaultSearchForm extends Component {
 
   static defaultProps = {
-    icons: defaultIcons,
+    icons: customIcons,
```
Just from a variable naming perspective, it feels wrong to set a default to a variable called `customIcons`.

landonreed commented on lib/components/narrative/narrative-routing-results.js:
```diff
@@ -32,12 +31,8 @@ class NarrativeRoutingResults extends Component {
-          customIcons={customIcons}
-        />
+      routingType === 'ITINERARY' &&
+        <TabbedItineraries itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} itineraries={itineraries} customIcons={customIcons} />
```
Probably warrants putting this on multiple lines (seems to violate the characters per line limit).

landonreed commented:
Also, I noticed a bug: when there are two car rental mode options and one is selected, both are highlighted:
![image](https://user-images.githubusercontent.com/2370911/80385064-9d4e6a80-8873-11ea-9e2c-1be836744c05.png)

`config.yml` (modes only):
```
  accessModes:
    - mode: BICYCLE
      label: Transit + Personal bike
    - mode: BICYCLE_RENT
      label: Transit + Biketown
      company: BIKETOWN
    - mode: MICROMOBILITY_RENT
      label: Transit + E-scooter rental
    - mode: CAR_PARK
      label: Park & Ride
    - mode: CAR_RENT
      label: Transit + car2go
      company: CAR2GO
    - mode: CAR_RENT
      label: Transit + REACH NOW
      company: REACHNOW
    - mode: CAR_HAIL
      label: Transit + Uber
      company: UBER
      showWheelchairSetting: true
  exclusiveModes:
    - WALK
    - BICYCLE
```

evansiroky commented on lib/components/narrative/line-itin/itin-summary.js:
```diff
@@ -82,6 +82,19 @@ export default class ItinerarySummary extends Component {
 
+  getIcon = (iconId, customIcons) => {
+    // Check if there is a custom icon
+    if (customIcons && iconId in customIcons) {
```
Is there a way we can get rid of this `customIcons` variable altogether? It would be nice to have everything inside the mode icon component.

binh-dam-ibigroup replied:
Working on it. I will shortly push a PR against OTP-ui to address that.

binh-dam-ibigroup replied:
This is resolved with opentripplanner/otp-ui#119.

binh-dam-ibigroup commented:
@landonreed The issues with mode selection you are mentioning above are handled by https://github.com/opentripplanner/otp-ui/pull/119.

landonreed commented on example.js:
```diff
@@ -90,14 +94,15 @@ class OtpRRExample extends Component {
     /** mobile view **/
     const mobileView = (
-      <MobileMain map={(<Map />)} title={(<div className='navbar-title'>OpenTripPlanner</div>)} />
+      <MobileMain LegIcon={MyLegIcon} ModeIcon={MyModeIcon} map={(<Map />)} title={(<div className='navbar-title'>OpenTripPlanner</div>)} />
```
Let's put this on multiple lines.

landonreed commented on lib/components/mobile/main.js:
```diff
@@ -99,10 +99,10 @@ class MobileMain extends Component {
 
       case MobileScreens.RESULTS_SUMMARY:
-        return <MobileResultsScreen map={map} itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} icons={icons} />
+        return <MobileResultsScreen map={map} itineraryClass={itineraryClass} itineraryFooter={itineraryFooter} LegIcon={LegIcon} />
```
multiple lines

landonreed commented on lib/components/mobile/results-screen.js:
```diff
@@ -95,7 +95,7 @@ class MobileResultsScreen extends Component {
 
   render () {
-    const { error, icons, itineraryClass, itineraryFooter, query, realtimeEffects, resultCount, useRealtime, activeItineraryIndex } = this.props
+    const { error, itineraryClass, itineraryFooter, LegIcon, query, realtimeEffects, resultCount, useRealtime, activeItineraryIndex } = this.props
```
multiple lines

landonreed commented on example.js:
```diff
@@ -26,6 +27,9 @@ import {
 import otpConfig from './config.yml'
 
+const MyLegIcon = TriMetLegIcon // For testing, you can use () => <Ferry />
+const MyModeIcon = () => TriMetModeIcon // For testing, you can use () => <AerialTram />
```
Can you add commented out lines below this that people can easily uncomment to try out the testing overrides? Also, it would be good to describe what the testing will show (and where).

landonreed commented on example.js:
```diff
@@ -27,8 +27,21 @@ import {
+ * For more advanced testing, you can replicate and customize components and observe the change in icons.
+ * - For LegIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-leg-icon.js
+ * - For ModeIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-mode-icon.js
+ */
```
Fantastico!

binh-dam-ibigroup commented:
@landonreed per a suggestion by @evansiroky, I removed the `customIcons` support but I realized we might still need some folks to occasionally override the std transit or streetcar icons, for instance without having to define a whole new `ModeIcon` or `LegIcon`. Let me think through it.

binh-dam-ibigroup commented:
@landonreed and @evansiroky, I have updated the opentripplanner/otp-ui packages and updated the examples for custom icon sets accordingly, so you can review.

landonreed commented on example.js:
```diff
@@ -28,21 +38,56 @@ import {
+  return <LegIcon leg={leg} ModeIcon={MyModeIcon} {...props} />
+}
+*/
+
```
@binh-dam-ibigroup, this is growing to be a lot to include in the `example.js` file. I think it might make more sense to move this to a `custom-icons.js` file and enable playing around with icon overrides with a single change. I've provided some changes in pursuit of this in https://github.com/opentripplanner/otp-react-redux/pull/164.

binh-dam-ibigroup replied:
Good idea.

evansiroky commented on lib/components/mobile/results-screen.js:
```diff
@@ -95,7 +95,17 @@ class MobileResultsScreen extends Component {
+      realtimeEffects,
+      resultCount,
+      useRealtime,
+      activeItineraryIndex
```
So close to perfect alphabetization until here.

landonreed commented:
:tada: This PR is included in version 1.0.0-beta.1 :tada:

The release is available on:
- [npm package (@beta dist-tag)](https://www.npmjs.com/package/otp-react-redux/v/1.0.0-beta.1)
- [GitHub release](https://github.com/opentripplanner/otp-react-redux/releases/tag/v1.0.0-beta.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

landonreed commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/otp-react-redux)
- [GitHub release](https://github.com/opentripplanner/otp-react-redux/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
